docs/src/dplyr.md

```diff
@@ -0,0 +1,293 @@
+
+```@example 1
+@rsubset msleep begin 
+    :sleep_total >= 16 
+    :bodywt >= 1
+end
+```
+
+Filter the rows for mammals in the Perissodactyla and Primates taxonomic order
+
+```@example 1
+@rsubset msleep :order in ["Perissodactyla", "Primates"]
+```
+
+You can use the boolean operators (e.g. >, <, >=, <=, !=, in) to create the logical tests. 
+
+# Chain.jl
+
+Before we go any further, let's introduce the `@chain` macro from [Chain.jl](https://github.com/jkrumbiegel/Chain.jl), which is re-exported by DataFramesMeta.jl. `@chain` allows for you to pipe the output from one operation into the input of another operation.  The idea of piping is to read the functions from left to right. The syntax and design of `@chain` is very similar to `%>%` which users of dplyr are familiar with.
+
+To show it's usefulness, let's use `@select` and `@rsubset` one after the other. 
+
+```
+msleep_1 = @select msleep :name :sleep_total
+msleep_2 = @rsubset msleep_1 :sleep_total > 16
+```
+
+Now in this case, we will pipe the msleep data frame to the function that will select two columns (name and sleep\_total) and then pipe the new data frame to the `@rsubset` opertaion. This method involves awkwardly creating and naming temporary data frames. This can be avoided with `@chain`.
+
+```@example 1
+@chain msleep begin 
+  @select :name :sleep_total
+  @rsubset :sleep_total > 16
+end
+```
+
+You will soon see how useful the `@chain` macro is when we start to combine many functions.  
+
+# Back To dplyr Verbs In Action
+
+Now that you know about the `@chain` macro, we will use it throughout the rest of this tutorial. 
+
+## Arrange Or Re-order Rows Using `@orderby`
+
+To arrange (or re-order) rows by a particular column, such as the taxonomic order, list the name of the column you want to arrange the rows by:
+
+```@example 1
+@orderby msleep :order
+```
+
+Now we will select three columns from msleep, arrange the rows by the taxonomic order and then arrange the rows by sleep\_total. Finally, keep the first 10 rows of the data frame.
+
+```@example 1
+@chain msleep begin 
+    @select :name :order :sleep_total
+    @orderby :order :sleep_total
+    first(10)
+end
+```
+
+Same as above, except here we filter the rows for mammals that sleep for 16 or more hours, instead of showing the head of the final data frame:
+
+```@example 1
+@chain msleep begin 
+    @select :name :order :sleep_total
+    @orderby :order :sleep_total 
+    @rsubset :sleep_total > 16
+end
+```
+
+Something slightly more complicated: same as above, except arrange the rows in the `:sleep_total` column in a descending order. For this, use the function `sortperm` with the keyword argument `rev=true`.
+
+```@example 1
+@chain msleep begin 
+    @select :name :order :sleep_total
+    @orderby begin 
+        :order 
+        sortperm(:sleep_total, rev=true)
```

Comment:
is this really needed? there is no way to pass `rev` to `@orderby`?

Reply:
No, we don't have keyword arguments yet.